### PR TITLE
check all bookies of writeset are writable

### PR DESCRIPTION
**Pull request by wuYin**
### Motivation

#1088 introduced ensemble writable checking before sending requests, but we should check bookies of writeset, instead of the first few bookies in current ensemble.

### Changes

Get the bookies of writeset from ensemble and check writeable.

Related change: https://github.com/apache/bookkeeper/pull/1088/files#diff-1d893bb31553b5e1f55c8301d04ae15f38e0d35f531f9dd22475128b7972ddf9R1108

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java [hunk 1]
```diff
@@ -1224,7 +1224,9 @@ private boolean isWriteSetWritable(DistributionSchedule.WriteSet writeSet,
         int nonWritableCount = 0;
         List<BookieId> currentEnsemble = getCurrentEnsemble();
         for (int i = 0; i < sz; i++) {
-            if (!clientCtx.getBookieClient().isWritable(currentEnsemble.get(i), ledgerId)) {
+            int writeBookieIndex = writeSet.get(i);
+            if (writeBookieIndex < currentEnsemble.size() &&
+                !clientCtx.getBookieClient().isWritable(currentEnsemble.get(writeBookieIndex), key)) {
```
Comment (dlg99) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java:1229]:
```suggestion
                !clientCtx.getBookieClient().isWritable(currentEnsemble.get(writeBookieIndex), ledgerId)) {
```

**Comment by dlg99** [comment]
@wuYin please fix compile error and consider adding unit test
```
/home/runner/work/bookkeeper/bookkeeper/bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java:1229: error: cannot find symbol

                !clientCtx.getBookieClient().isWritable(currentEnsemble.get(writeBookieIndex), key)) {
                                                                                               ^
  symbol:   variable key
  location: class LedgerHandle
> Task :bookkeeper-server:compileJava
1 error


> Task :bookkeeper-server:compileJava FAILED

```

**Comment by dlg99** [comment]
@wuYin are you working on the tests? I'd love to include this into release 4.15 with appropriate tests

**Comment by wuYin** [comment]
@dlg99 
Sorry for the late reply, now I added writeset writeable check, this test covered the bug, but not involved back-pressure.
Thank you for your patience.

**Pull request merged**